examples/ramberg_osgood.py

```diff
@@ -0,0 +1,666 @@
+                self._stress.append(df.fem.Function(QV_sub))
+
+                # ### submesh and subspace for tanget
+                QT_parent, QT_child, QT_sub = build_view(cells, QT)
+                self.QT_views.append((QT_parent, QT_child, QT_sub))
+                self._tangent.append(df.fem.Function(QT_sub))
+
+        self.stress = df.fem.Function(QV)
+        self.tangent = df.fem.Function(QT)
+
+        u_, du = ufl.TestFunction(u.function_space), ufl.TrialFunction(u.function_space)
+
+        self.metadata = {"quadrature_degree": q_degree, "quadrature_scheme": "default"}
+        self.dxm = ufl.dx(metadata=self.metadata)
+
+        eps = self.eps
+        self.R_form = ufl.inner(eps(u_), self.stress) * self.dxm
+        self.dR_form = ufl.inner(eps(du), ufl.dot(self.tangent, eps(u_))) * self.dxm
+        self.u = u
+
+        basix_celltype = getattr(basix.CellType, mesh.topology.cell_type.name)
+        self.q_points, _ = basix.make_quadrature(basix_celltype, q_degree)
+        self.strain_expr = df.fem.Expression(eps(u), self.q_points)
+
+    def compile(self, bcs):
+        # compile should have same args as super().__init__ !!
+        # otherwise no control over how to compile the ufl form
+
+        R = self.R_form
+        u = self.u
+        dR = self.dR_form
+        super().__init__(R, u, bcs=bcs, J=dR)
+
+    def form(self, x: PETSc.Vec):
+        """This function is called before the residual or Jacobian is
+        computed. This is usually used to update ghost values.
+        Parameters
+        ----------
+        x
+            The vector containing the latest solution
+        """
+        super().form(x)
+
+        # FIXME
+        time = 0.0
+        history = np.array([])
+
+        for k, (law, cells) in enumerate(self.laws):
+            with df.common.Timer("strain_evaluation"):
+                self.strain_expr.eval(
+                    cells, self._strain[k].x.array.reshape(cells.size, -1)
+                )
+
+            with df.common.Timer("stress_evaluation"):
+                law.evaluate(
+                    time,
+                    self._strain[k].x.array,
+                    self._stress[k].x.array,
+                    self._tangent[k].x.array,
+                    history,
+                )
+
+            with df.common.Timer("stress-local-to-global"):
+                parent, child, _ = self.QV_views[k]
+                stress_global = self.stress.x.array.reshape(-1, 6)
+                stress_local = self._stress[k].x.array.reshape(-1, 6)
+                stress_global[parent] = stress_local[child]
+                c_parent, c_child, _ = self.QT_views[k]
+                tangent_global = self.tangent.x.array.reshape(-1, 36)
+                tangent_local = self._tangent[k].x.array.reshape(-1, 36)
+                tangent_global[c_parent] = tangent_local[c_child]
+
+        self.stress.x.scatter_forward()
+        self.tangent.x.scatter_forward()
+
+    def eps(self, u):
```

Comment:
Something like
```python
from .stress_strain import ufl_mandel_strain
```
inside `interfaces.py` will lead to a circular import.

My current solution is to let `IncrSmallStrainProblem.eps` be an `abstractmethod` that the user has to implement. I would consider this the safer option. If there was some default `eps` and the user does not have to think about it, then this could potentially lead to the `IncrSmallStrainProblem` computing _mandel strain_, although user has implemented some material routine expecting standard voigt basis.